### PR TITLE
Add prettier configuration and pre-commit hook

### PR DESCRIPTION
**Pull request by griffbrad**
I'm taking the current pre-commit hook and prettier config from Calypso and modifying them slightly to work for dserve.  Code style is pretty inconsistent in dserve currently, so hopefully this will allow us to tidy up a bit.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,7 +50,8 @@
 		"serve": "node build/index.js",
 		"serve:forever": "node build/daemon.js",
 		"build-ts": "tsc",
-		"watch-ts": "tsc -w"
+		"watch-ts": "tsc -w",
+		"reformat-files": "./node_modules/.bin/prettier --write --require-pragma \"src/**/*.{ts,tsx,scss}\""
```
Comment (samouri) [inline package.json]: the whole codebase is prettified, or at least it should be.  I believe you can remove the `--require-pragma`

**Comment by blowery** [comment]
@griffbrad reworked this. thoughts?

**Comment by sirreal** [comment]
I added #87 which contains current Calypso prettier integration including the optimized prettier bundle.

Do we need/want a pre-commit hook on this project given that few folks work on this with any regularity?

**Comment by blowery** [comment]
Closing this for now since we have #87

**Pull request closed without merging**